jxl/src/frame/modular/transforms.rs

```diff
@@ -272,18 +272,7 @@ pub fn make_grids(
                     );
                     let offsets = match predictor {
                         Predictor::Zero => [].as_slice(),
-                        Predictor::West | Predictor::WestWest => &[(-1, 0)],
-                        Predictor::North => &[(0, -1)],
-                        Predictor::Select
-                        | Predictor::Gradient
-                        | Predictor::NorthWest
-                        | Predictor::AverageWestAndNorth
-                        | Predictor::AverageWestAndNorthWest
-                        | Predictor::AverageNorthAndNorthWest => &[(0, -1), (-1, 0), (-1, -1)],
-                        Predictor::NorthEast | Predictor::AverageNorthAndNorthEast => {
-                            &[(0, -1), (1, 0), (1, -1)]
-                        }
-                        Predictor::AverageAll | Predictor::Weighted => unreachable!(),
+                        _ => &[(0, -1), (-1, 0), (-1, -1)],
```

Comment:
Why do we need all of those even for West/WestWest/North?

Reply:
Even West can use Top if x==0, maybe we could spare (-1, -1) sometimes but that would make the delta palette code a bit more complex (now we always use a rectangle of groups).